homework/82197/js2html/Lexer.h

```diff
@@ -0,0 +1,427 @@
+			return ProduceSingleCharacterToken(type, "||");
+		case TokenType::StarEqual:
+			return ProduceSingleCharacterToken(type, "*=");
+		case TokenType::StarStarEqual:
+			return ProduceSingleCharacterToken(type, "**=");
+		case TokenType::DivisionEqual:
+			return ProduceSingleCharacterToken(type, "/=");
+		case TokenType::ModuloEqual:
+			return ProduceSingleCharacterToken(type, "%=");
+		case TokenType::PlusEqual:
+			return ProduceSingleCharacterToken(type, "+=");
+		case TokenType::MinusEqual:
+			return ProduceSingleCharacterToken(type, "-=");
+		case TokenType::LeftShiftEqual:
+			return ProduceSingleCharacterToken(type, "<<=");
+		case TokenType::RightShiftEqual:
+			return ProduceSingleCharacterToken(type, ">>=");
+		case TokenType::UnsignedRightShiftEqual:
+			return ProduceSingleCharacterToken(type, ">>>=");
+		case TokenType::BitwiseAndEqual:
+			return ProduceSingleCharacterToken(type, "&=");
+		case TokenType::BitwiseXorEqual:
+			return ProduceSingleCharacterToken(type, "^=");
+		case TokenType::BitwiseOrEqual:
+			return ProduceSingleCharacterToken(type, "|=");
+		case TokenType::Backslash:
+			return ProduceSingleCharacterToken(type, "\\");
+		case TokenType::NewLine:
+			return ProduceToken(TokenType::NewLine, "<br></br>");
+		case TokenType::Whitespace: 
+			return ProduceToken(TokenType::Whitespace, "&nbsp;");
+		case TokenType::Tab: 
+			return ProduceToken(TokenType::Tab, "&nbsp;&nbsp;&nbsp;&nbsp;");
+		default: // EoF Invalid
+			break;
+		}
+		return ProduceToken(type, "");
+	}
+	Token ProduceCommentToken(TokenType type, const IPLString& comment)
+	{
+		unsigned int lastTokenPosition = m_LastTokenPozition;
+		m_LastTokenPozition = m_Column;
+		return Token { type, m_Line, lastTokenPosition, "<p style=\"display:inline; color:green\">" + comment + "</p>", 0.0};
+	}
+	Token ProduceSingleCharacterToken(TokenType type, const IPLString& operatorLexeme)
+	{
+		unsigned int lastTokenPosition = m_LastTokenPozition;
+		m_LastTokenPozition = m_Column;
+		return Token{type, m_Line, lastTokenPosition, "<p style=\"display:inline; color:yellow\">" + operatorLexeme + "</p>", 0.0};
+	}
+	Token ProduceStringToken(TokenType type, const IPLString& str)
+	{
+		unsigned int lastTokenPosition = m_LastTokenPozition;
+		m_LastTokenPozition = m_Column;
+		return Token{type, m_Line, lastTokenPosition, "<p style=\"display:inline; color:purple\">" + str + "</p>", 0.0};
+	}
+	Token ProduceIdentifierToken(TokenType type, const IPLString& identifier)
+	{
+		unsigned int lastTokenPosition = m_LastTokenPozition;
+		m_LastTokenPozition = m_Column;
+		return Token{type, m_Line, lastTokenPosition, "<p style=\"display:inline; color:#FFF59E\">" + identifier + "</p>", 0.0};
+	}
+	Token ProduceKeywordToken(TokenType type, const IPLString& keyword)
+	{
+		unsigned int lastTokenPosition = m_LastTokenPozition;
+		m_LastTokenPozition = m_Column;
+		return Token{type, m_Line, lastTokenPosition, "<p style=\"display:inline; color:blue\">" + keyword + "</p>", 0.0};
```

Comment:
Could have let the tokens be tokens without knowledge of the HTML output and add the styling in main.cpp
It would be nice to keep the original formatting of the code file in the html.